crates/sui-graphql-rpc/src/types/query.rs

```diff
@@ -239,3 +350,89 @@ impl Query {
             .extend()
     }
 }
+
+// Intermediate enum to handle the two possible types of `tx_bytes` in `dry_run_transaction_block` usiing FN
+enum DryRunExecArg {
+    TransactionData(TransactionData),
+    TransactionKindWithMeta {
+        kind: TransactionKind,
+        sender: NativeSuiAddress,
+        gas_price: Option<u64>,
+        epoch: Option<u64>,
+        gas_objects: Option<Vec<NativeSuiAddress>>,
+    },
+}
+
+impl DryRunExecArg {
+    pub fn into_transaction_data(self) -> TransactionData {
+        match self {
+            DryRunExecArg::TransactionData(td) => return td,
+
+            // We hit this scenario if the TransactionKind and meta are provided
+            // and skip_checks is false.
+            // This requires forming a full TransactionData object.
+            DryRunExecArg::TransactionKindWithMeta {
+                kind,
+                sender,
+                gas_price,
+                epoch,
+                gas_objects,
+            } => {
+                // TODO: We need to handle the gas data
+
+                // We dont have enough info for these
+                // let gas_payment: ObjectRef;
+                // let gas_budget: u64;
+                TransactionData::new_with_epoch(
```

Comment:
Where do we get all this GasData requirements from?

Reply:
Paraphrasing from our out-of-band chat:

This one was an oversight on my part, in the design -- I think gas objects need to be passed in as a full ObjectRef, and `TransactionMetadata` should include a gas budget too.

Reply:
It sounds like you're hitting up against something specific, can you share details on that, so far the conversions are `unimplemented!()` so that's not much to go by -- we can also chat about it when we meet later on. The draft GraphQL schema is not necessarily designed to map to the way fullnode responds to dry-run result today, so there is some shoe-horning to be expected.